### PR TITLE
refactor(headcount): change converting to afk check

### LINKED ISSUE #137: Starting AFK Check should end Headcount
If someone (Blakstone) starts an afk check through a command, and not by ending the headcount, the headcount should end.

### LINKED ISSUE #139: Weird AFK check started time on control panel
Looks like starting an AFK check from the control panel (in raid mode) gives a weird time for when the AFK check started at.

image

### LINKED ISSUE #151: The bot should edit the headcount message so that it says that the headcount has been converted to AFK check ONLY when the AFK check actually starts
Right now, the bot will edit the headcount embed to indicate that it has been converted to an AFK check even before the RL selects a VC for the soon-to-be-converted AFK check, giving the impression that the bot might be super delayed.

### PR DESCRIPTION
**Pull request by nyapat**
- AFK check relies on current time - instead of using the voice channel's creation time. closes #139
- fun workaround, starting a new AFK check using the slash command checks for any ActiveHeadcounts with the same user id and silently converts instead of creating an AFK check. closes #137
- adds a new enum to show users that the headcount is converting, improving ux. closes #151

small PR per time dman crashes my run - I'll handle rebasing if you want

### CODE REVIEW

**Comment by dmansdman** [comment]
Re: https://github.com/ewang2002/ToogaBooga/issues/137 Not a good solution.  A person may choose to start an AFK for a different dungeon than the one that they put up a headcount for.  I can see this happening for event leaders who lead all sorts of stuff.  I recommend simply aborting existing headcounts rather than converting.

File: src/commands/raid-leaders/StartAfkCheck.ts [hunk 1]
```diff
@@ -121,8 +122,17 @@ export class StartAfkCheck extends BaseCommand {
...
         }
+        
+        // Step 5: Check if there's a current headcount running. This is structured here for easier selections by the RL.
+        
+        const activeHeadcount = HeadcountInstance.ActiveHeadcounts.find(headcount => headcount.getHeadcountInfoObject()!.memberInit == ctx.interaction.user.id)
 
-        // Step 5: Start it
+        if (activeHeadcount) {
+            await activeHeadcount.convertHeadcount(ctx.member!, vcToUse);
+            return 0;
```
Comment (dmansdman) [inline src/commands/raid-leaders/StartAfkCheck.ts:132]: See previous comment.  Would change this to abort and not return 0.  Also consider the case that there are multiple active headcounts up, would abort ALL of them.

**Review by dmansdman — DISMISSED** [review]
Pending items for approval:
1) 137 adjustment
2) testing

Besides minor adjustment to 137, the other changes look good.  Ensure you have tested locally on the test server, I haven't noticed activity there regarding these changes.

**Comment by ewang2002** [comment]
@dmansdman do you want to look over this at some point again?

**Review by ewang2002 — APPROVED** [review]
Functionality-wise (from doing some testing), and code-wise, this looks fine as is. Nice!

**Pull request merged**